Setuptools 60.X breaks action-ros-ci

## Description
Setuptools 60 has switched back to their version of distutils(https://github.com/pypa/setuptools/pull/2896). Therefore running `python setup.py install` with the argument `--install-layout=deb` fails. Because the setuptools version of distutils doesn't have the `--install-layout` argument.

`python setup.py install` is trigger in https://github.com/ros/catkin/blob/noetic-devel/cmake/templates/python_distutils_install.sh.in, which is trigger by `colcon build --event-handlers console_cohesion+ --symlink-install`

### Expected Behavior
Just performing the install procedure

### Actual Behavior
```
/usr/bin/env PYTHONPATH=/home/runner/work/rosbridge_suite/rosbridge_suite/ros_ws/install/rosbridge_library/lib/python3/dist-packages:/home/runner/work/rosbridge_suite/rosbridge_suite/ros_ws/build/rosbridge_library/lib/python3/dist-packages:/opt/ros/noetic/lib/python3/dist-packages CATKIN_BINARY_DIR=/home/runner/work/rosbridge_suite/rosbridge_suite/ros_ws/build/rosbridge_library /usr/bin/python3 /home/runner/work/rosbridge_suite/rosbridge_suite/ros_ws/src/04qw58xlp1nu/rosbridge_suite/rosbridge_library/setup.py build --build-base /home/runner/work/rosbridge_suite/rosbridge_suite/ros_ws/build/rosbridge_library install --root=/ --install-layout=deb --prefix=/home/runner/work/rosbridge_suite/rosbridge_suite/ros_ws/install/rosbridge_library --install-scripts=/home/runner/work/rosbridge_suite/rosbridge_suite/ros_ws/install/rosbridge_library/bin
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: option --install-layout not recognized
  CMake Error at catkin_generated/safe_execute_install.cmake:4 (message):
    
    execute_process(/home/runner/work/rosbridge_suite/rosbridge_suite/ros_ws/build/rosbridge_library/catkin_generated/python_distutils_install.sh)
    returned error code
```

## System (please complete the following information)
 - OS: Ubuntu 20.04
 - ROS Distro: Noetic

## Additional context
I have created the issue here. As this action installs setuptools. You also chose to especially install setuptools with pip with `--upgrade` to overrule the debian version.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -42,7 +42,6 @@ jobs:
         docker_image:
           - "ubuntu:bionic"
           - "ubuntu:focal"
-          - "ubuntu:xenial"
     container:
       image: ${{ matrix.docker_image }}
     steps:
@@ -97,7 +96,6 @@ jobs:
       fail-fast: false
       matrix:
         ros_distribution:
-          - kinetic
           - melodic
           - noetic
           - dashing
@@ -113,11 +111,6 @@ jobs:
         #
         # Platforms are defined in REP 3: https://ros.org/reps/rep-0003.html
         include:
-          # Kinetic Kame (May 2016 - May 2021)
-          - docker_image: ubuntu:xenial
-            ros_distribution: kinetic
-            ros_version: 1
-
           # Melodic Morenia (May 2018 - May 2023)
           - docker_image: ubuntu:bionic
             ros_distribution: melodic

--- a/src/package_manager/pip.ts
+++ b/src/package_manager/pip.ts
@@ -53,7 +53,7 @@ const pip3Packages: string[] = [
 	"pytest-repeat",
 	"pytest-rerunfailures",
 	"pytest-runner",
-	"setuptools",
+	"setuptools<60.0",
 	"wheel",
 ];
 

